Babylon/utils/macro.py

```diff
@@ -0,0 +1,92 @@
+import json
+import logging
+from typing import Any
+from typing import Optional
+from typing import Callable
+from pathlib import Path
+from time import sleep
+
+from .response import CommandResponse
+from .command_helper import run_command
+from .environment import Environment
+
+logger = logging.getLogger("Babylon")
+
+
+class Macro():
+    """Handles Macro Command Chaining
+    """
+    STATUS_OK = 0
+    STATUS_ERROR = 1
+
+    def __init__(self, name: str):
+        self.name = name
+        self._responses: list[CommandResponse] = []
+        self.env = Environment()
+        self._status = self.STATUS_OK
+
+    def step(self,
+             command_line: list[str],
+             optional: bool = False,
+             store_at: Optional[str] = None,
+             run_if: bool = True) -> "Macro":
```

Comment:
Same comment here: it is not easy to understand what the parameter "run_if" does exactly at the first sight...

Reply:
Oh, I thought it was alright. I will document in docstring it as I don't know how to name it better.